experimental/generation/generator/src/library/dialogGenerator.ts

```diff
@@ -530,6 +531,22 @@ async function generateSingleton(schema: string, inDir: string, outDir: string)
     }
 }
 
+async function generateQnA(schema: string, allLocales: string[], inDir: string) {
+    let files = fs.readdirSync(inDir)
+    for (let locale of allLocales) {
+        let qnaText = ''
+        for (let file of files) {
+            if (file.endsWith('qna')) {
+                let content = await fs.readFile(ppath.join(inDir, file), 'utf-8')
+                qnaText = qnaText + os.EOL + content.toString().replace(ReplaceGeneratorPattern, '')
+                await fs.unlink(ppath.join(inDir, file))
+            }
+        }
+        let qnaName = `${schema}.${locale}.qna`
+        qnaText += `${os.EOL}> Generator: ${computeHash(qnaText)}`
+        await fs.writeFile(ppath.join(`${inDir}/${locale}`, qnaName), qnaText)
+    }
+}
```

Comment:
Maybe I'm missing something, but I don't think you need this at all.  It should all be done through templates.